pkgs/development/tools/misc/cli11/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation rec {
+  name = "cli11-${version}";
+  version = "1.7.1";
+
+  src = fetchFromGitHub {
+    owner = "CLIUtils";
+    repo = "CLI11";
+    rev = "v${version}";
+    sha256 = "0wddck970pczk7c201i2g6s85mkv4f2f4zxy6mndh3pfz41wcs2d";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  cmakeFlags = [
+    "-DCLI11_TESTING=OFF"
```

Comment:
`doCheck = false` due to not able to have the code find gtest, despite trying several things (comments in code)

Reply:
Hmm indeed this gtest thing is tricky. Here is how I got it right:
```diff
diff --git a/pkgs/development/tools/misc/cli11/default.nix b/pkgs/development/tools/misc/cli11/default.nix
index 87b409562b7..bd735373539 100644
--- a/pkgs/development/tools/misc/cli11/default.nix
+++ b/pkgs/development/tools/misc/cli11/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchFromGitHub, cmake, gtest }:
+{ stdenv, fetchFromGitHub, cmake, python, gtest, boost }:
 
 stdenv.mkDerivation rec {
   name = "cli11-${version}";
@@ -15,23 +15,12 @@ stdenv.mkDerivation rec {
     cmake
   ];
 
-  checkInputs = [
-    gtest
-  ];
-
-
-  # NOTE: disabling tests due to gtest not found issue
-  doCheck = false;
-  cmakeFlags = [
-    "-DCLI11_TESTING=OFF"
-  #  "-DNO_GTEST=ON"
-  #  "-DGTEST_ROOT=${gtest.src}/googletest"
-  #  "-DGTEST_INCLUDE_DIR=${gtest.dev}/include"
-  #  "-DGTEST_HOME=${gtest}"
-  #  "-DCMAKE_CURRENT_BINARY_DIR=${gtest.dev}/bin"
-  #  "-DBUILD_SHARED_LIBS=ON"
-  #  "-DGTEST_LIB_DIR=${gtest.dev}/lib"
-  ];
+  doCheck = true;
+  preConfigure = stdenv.lib.optionalString doCheck ''
+    rm -rf extern/googletest
+    ln -sf ${gtest.src} extern/googletest
+  '';
+  checkInputs = [ boost python ];
 
   enableParallelBuilding = true;
```
The symlink is necessary because of https://github.com/CLIUtils/CLI11/blob/master/tests/CMakeLists.txt#L1

Reply:
fixed